db/flush_job_test.cc

```diff
@@ -487,6 +507,137 @@ TEST_F(FlushJobTest, Snapshots) {
   job_context.Clean();
 }
 
+class FlushJobTimestampTest : public FlushJobTestBase {
+ public:
+  FlushJobTimestampTest()
+      : FlushJobTestBase(test::PerThreadDBPath("flush_job_ts_gc_test"),
+                         test::ComparatorWithU64Ts()) {}
+
+  void AddKeyValueToMemtable(MemTable* memtable, Slice key, uint64_t ts,
+                             SequenceNumber seq, ValueType value_type,
+                             Slice value) {
+    std::string key_str(key.data(), key.size());
+    std::string ts_str;
+    PutFixed64(&ts_str, ts);
+    key_str.append(ts_str);
```

Comment:
I think you could encode directly into `key_str` here (no need for a separate `ts_str`).